src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/srm/algebras/BDD.cs

```diff
@@ -7,18 +7,34 @@
 namespace System.Text.RegularExpressions.SRM
 {
     /// <summary>
-    /// Represents a Binary Decision Diagram.
+    /// Represents nodes in a Binary Decision Diagram (BDD), which compactly represent sets of integers. All non-leaf
+    /// nodes have an Ordinal, which indicates the position of the bit the node relates to (0 for the least significant
+    /// bit), and two children, One and Zero, for the cases of the current bit being 1 or 0, respectively. An integer
+    /// belongs to the set represented by the BDD if the path from the root following the branches that correspond to
+    /// the bits of the integer leads to the True leaf. This class also supports multi-terminal BDDs, i.e. ones where
```

Comment:
Nit: later on you abbreviate multi-terminal BDDs as MTBDD, so consider adding a `(MTBDD)` parenthetical here.
```suggestion
    /// the bits of the integer leads to the True leaf. This class also supports multi-terminal BDDs (MTBDD), i.e. ones where
```